frontend/src/views/Transactions.vue

```diff
@@ -1,100 +1,123 @@
-            v-for="option in typeFilter.options"
-            :key="option.value"
-            :value="option.value"
+      <div class="flex gap-2">
+        <SearchField
+          v-model="search"
+        />
+        <label>
+          <select
+            id="type"
+            v-model="typeFilter.value"
+            name="type"
+            class="rounded-lg border-0"
           >
-            {{ option.display }}
-          </option>
-        </select>
-      </label>
-    </div>
+            <option
+              v-for="option in typeFilter.options"
+              :key="option.value"
+              :value="option.value"
+            >
+              {{ option.display }}
+            </option>
+          </select>
+        </label>
+      </div>
+    </HeadlineActions>
 
-    <DataTable :config="data">
+    <DataTable
+      :config="config"
+      :rows="rows"
+    >
       <template
         #position="{ row }"
       >
         <AppLink
-          :ticker="row['ticker']"
+          :ticker="row.symbol"
           to="ticker_details"
         >
-          {{ row['position'] }}
+          {{ row.name }}
         </AppLink>
       </template>
       <template
         #type="{ row }"
       >
         <div
+          :class="row ==='BUY'? 'text-blue-500' : 'text-yellow-500'"
           class="flex gap-2"
-          :class="row['type'] ==='BUY'? 'text-blue-500' : 'text-yellow-500'"
         >
-          <svg
-            xmlns="http://www.w3.org/2000/svg"
+          <ArrowCircleLeftIcon
             class="h-6 w-6"
-            :class="row['type'] ==='BUY'? 'transform rotate-180' : ''"
-            viewBox="0 0 20 20"
-            fill="currentColor"
-          >
-            <path
-              fill-rule="evenodd"
-              d="M10 18a8 8 0 100-16 8 8 0 000 16zm.707-10.293a1 1 0 00-1.414-1.414l-3 3a1 1 0 000 1.414l3 3a1 1 0 001.414-1.414L9.414 11H13a1 1 0 100-2H9.414l1.293-1.293z"
-              clip-rule="evenodd"
-            />
-          </svg>
-          {{ $t(data.settings.translationPrefix + '.' + row['type'].toLowerCase()) }}
+            :class="row ==='BUY'? 'transform rotate-180' : ''"
+          />
+
+          {{ $t(config.settings.translationPrefix + '.' + row.toLowerCase()) }}
         </div>
       </template>
     </DataTable>
   </div>
 </template>
 
 <script lang="ts">
-import {computed, defineComponent, onMounted, onUnmounted, reactive} from 'vue'
+import {computed, defineComponent, onBeforeMount, onUnmounted, reactive, ref, toRefs} from 'vue'
 import {Transaction} from '../models'
 import dayjs from 'dayjs'
-import DataTable, {TableCellType, TableConfig, TableRow} from '../components/DataTable.vue'
+import DataTable from '../components/DataTable.vue'
 import AppLink from '../components/router/AppLink.vue'
+import {ArrowCircleLeftIcon} from '@heroicons/vue/solid'
 import HeadlineActions from '../components/HeadlineActions.vue'
+import SearchField from '../components/base/SearchField.vue'
 
 enum Column {
```

Comment:
We could probably get rid of this too. There is not really a need for a constant since the label is longer than its value, and it's fairly straightforward.